Quorum/apis/llms/llm_conversation.py

```diff
@@ -0,0 +1,105 @@
+from enum import StrEnum, auto
+from pathlib import Path
+
+class MessageRole(StrEnum):
+    USER = auto()
+    ASSISTANT = auto()
+
+
+MAX_TOKENS = 1024
+
+
+def tmp_send_message_to_claude(client, model, messages):
+    try:
+        response = client.messages.create(
+            max_tokens=MAX_TOKENS,
+            model=model,
+            messages=messages
+        )
+    except:
+        raise
+
+    with open(Path(__file__).parent / 'ClaudeExecutionsNum.txt', 'r+') as f:
+        num = int(f.read())
+        f.seek(0)
+        f.write(str(num + 1))
+        f.truncate()
+
+    return response
+
+
+class LLMConversation:
+    '''
+    Description
+    -----------
+    At least GPT and Claude share the exact same API. Probably other LLMs use the same API as well,
+    just using different clients. This abstract class builds is expected to be inherited and build
+    the relevant client to be used with the same implementation of interacting with the LLM while 
+    maintaining a conversation context.
+    '''
+    def __init__(self):
+        '''
+        Summary
+        -------
+        Initialize a conversation manager for an LLM.
+        '''
+        self.client = None
+        self.model = None
+        self.messages: list[dict[str, str]] = []
+        
+    def add_message(self, role: str, content: str) -> None:
+        '''
+        Summary
+        -------
+        Add a message to the conversation history.
+        
+        Parameters
+        ----------
+        role : MessageRole
+            What role to use.
+        content : str
+            The message content.
+        '''
+        self.messages.append({
+            'role': str(role),
+            'content': content
```

Comment:
Can be encapsulate to `Message` object

Reply:
This isn't how the API works actually.
What sends the message to the LLM is `client.messages.create` which takes the parameter `messages: list[MessageParam]`. `MessageParam` is a typed dict with exactly the fields "content" and "role" which can be either "user" or "assistant". 
My point is that this is how the API already works with typed dicts so encapsulating this in a message object will be incompatible with the API.
I considered actually encapsulating it in this `MessageParam` object but this import will be Anthropic specific which is bad, even though OpenAI share the same API (they even credit OpenAI in the source code).

Reply:
Ahhh thanks for the clarifications.